### ISSUE TITLE
ModuleNotFoundError: No module named 'sklearn.linear_model.logistic'

### ISSUE THREAD

**Issue by AR-9191** [oaraque/moral-foundations#6]
sklearn is installed but this error is returned when running 'string_moral_values'

**Comment by oaraque** [comment]
I cannot find this line in the code: 
```python
sklearn.linear_model.logistic
```

Can you copy the error messages?

**Comment by AR-9191** [comment]
ModuleNotFoundError                       Traceback (most recent call last)
<ipython-input-7-aa5bef7a07b8> in <module>
      3 text = "PLS help #HASHTAG's family. No one prepares for this. They are in need of any assistance you can offer"
      4 
----> 5 moralstrength.string_moral_values(text)

~/opt/anaconda3/envs/emfd/lib/python3.7/site-packages/moralstrength/moralstrength.py in string_moral_values(text, model)
    219     if model not in models:
    220         raise ValueError('Invalid model "{}" specified. Valid models are: {}'.format(moral,models))
--> 221     return {moral: estimate(text=text, moral=moral, model=model)[0] for moral in moral_options_predictions}
    222 
    223 # def file_moral_value(filename,trait,model):

~/opt/anaconda3/envs/emfd/lib/python3.7/site-packages/moralstrength/moralstrength.py in <dictcomp>(.0)
    219     if model not in models:
    220         raise ValueError('Invalid model "{}" specified. Valid models are: {}'.format(moral,models))
--> 221     return {moral: estimate(text=text, moral=moral, model=model)[0] for moral in moral_options_predictions}
    222 
    223 # def file_moral_value(filename,trait,model):

~/opt/anaconda3/envs/emfd/lib/python3.7/site-packages/moralstrength/estimators.py in estimate(text, moral, model, from_file)
     85 
     86     estimators, transformers = select_processes(model, moral)
---> 87     lr, transformers = load_models(estimators, transformers, moral)
     88 
     89     X = []

~/opt/anaconda3/envs/emfd/lib/python3.7/site-packages/moralstrength/estimators.py in load_models(estimators, transformers, moral)
     44 def load_models(estimators, transformers, moral):
     45     model_name = generate_model_name(estimators)
---> 46     lr = load_model(pck_name(model_name, moral))
     47 
     48     trans = {}

~/opt/anaconda3/envs/emfd/lib/python3.7/site-packages/moralstrength/estimators.py in load_model(model_name)
     39     full_path = os.path.join(dir_path, model_path)
     40     with open(full_path, 'rb') as f:
---> 41         m = pickle.load(f)
     42     return m
     43 

ModuleNotFoundError: No module named 'sklearn.linear_model.logistic'

**Comment by AR-9191** [comment]
Thanks for the help. The 'estimate_morals' function works but not this one.

**Comment by oaraque** [comment]
Yes, I see the problem now. The issue is that the pickled models are older than the current sklearn version, and there have been changes to the API.

For the moment, the error can be fixed running the code on an older sklearn version.

As a fix, the pickled models need to be updated to the new version, which requires retraining all of them. I will do that, but it will take a while.

I recommend changing the sklear version if the trained models are needed.

**Comment by AR-9191** [comment]
I changed to sklearn version 20.0 and now it works. Thank you

**Comment by oaraque** [comment]
Fixed issue with commit 19be687.
